### PR TITLE
Update AwsClusterTemplate printcolumn and remove unnecessary RBAC files

### PR DESCRIPTION
**Pull request by pydctw**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
1) Added printcolumn, `Age`, to AWSClusterTemplate. This is in line with other templates like KubeadmControlplaneTemplate or DockerClusterTemplate in CAPI.
2) Deleted unnecessary RBAC roles created for AWSClusterTemplate

Also, cleaned up a duplicate kubebuilder marker for AWSManagedControlPlane

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by pydctw** [comment]
cc @ykakarap. Removing the RBAC files as part of this PR once you give ok.

**Comment by Ankitasw** [comment]
Hey @pydctw , thanks for the PR.
Could you give some context for why this Age column is needed?

> Added printcolumn, Age, to AWSClusterTemplate

**Comment by pydctw** [comment]
> Hey @pydctw , thanks for the PR. Could you give some context for why this Age column is needed?
> 
> > Added printcolumn, Age, to AWSClusterTemplate

This is following the convention of newly added templates for ClusterClass in CAPI. Making them consistent.
* [KCPTemplate](https://github.com/kubernetes-sigs/cluster-api/blob/main/controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go#L33)
* [DockerClusterTemplate](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/docker/api/v1beta1/dockerclustertemplate_types.go#L32)

**Review by ykakarap — COMMENTED** [review]
+1 to dropping these RBAC files. We dont use them and can be dropped safely.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml [hunk 1]
```diff
@@ -342,7 +342,12 @@ spec:
         type: object
     served: true
     storage: false
-  - name: v1beta1
+  - additionalPrinterColumns:
+    - description: Time duration since creation of AWSClusterTemplate
+      jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    name: v1beta1
```
Comment (Ankitasw) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml:350]:
```suggestion
    - name: v1beta1
```

File: config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml [hunk 2]
```diff
@@ -696,6 +701,7 @@ spec:
         type: object
     served: true
     storage: true
+    subresources: {}
```
Comment (Ankitasw) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml:704]: Not sure how is it used, but in other templates I see it as below
```suggestion
   subresources:
      status: {}
```

Comment (pydctw) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml:704]: Hi @Ankitasw, this is automatically generated file and cannot be modified.

File: config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml (hunk 1 above)
Comment (pydctw) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_awsclustertemplates.yaml:350]: same. automatically generated.

**Pull request merged**